### PR TITLE
task1, zombies counters

### PR DESCRIPTION
**Pull request by Stereotip42**

### CODE REVIEW

File: js/app.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// Selectors for counters
+const TOTAL = 'total-zombies';
+const DEAD = 'dead-zombies';
+const SMALL = 'small-zombies';
+const MAD = 'mad-zombies';
+const STRONG = 'strong-zombies';
+
+let counters = {
```
Comment (ExSamaelSolbeg) [inline js/app.js]: const

File: js/app.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    [TOTAL]: 0,
+    [DEAD]: 0,
+    [SMALL]: 0,
+    [MAD]: 0,
+    [STRONG]: 0,
+};
+
+let calculateZombies = () => {
```
Comment (ExSamaelSolbeg) [inline js/app.js]: 1) const
2) what's the point of defining a function? it's just a one time call

please implement a solution without any function at all

File: js/app.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    let totalCounter = 0;
+
+    zombiesData.map((zombieHP) => {
+        let zombieType = getZombieType(zombieHP);
+        let counter = counters[zombieType];
+
+        counters[zombieType] = ++counter;
```
Comment (ExSamaelSolbeg) [inline js/app.js]: please do not use pre-increment
what's the point of using 

```
let counter = counters[zombieType];
counters[zombieType] = ++counter;

```
?

you just can use simple increment like "anyCounter++"

File: js/app.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    [DEAD]: 0,
+    [SMALL]: 0,
+    [MAD]: 0,
+    [STRONG]: 0,
+};
+
+let calculateZombies = () => {
+    let totalCounter = 0;
+
+    zombiesData.map((zombieHP) => {
```
Comment (ExSamaelSolbeg) [inline js/app.js]: map returns a new array
so it doesn't make sense to use "map" in this case
it's just a simple "for"

File: js/app.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        totalCounter++;
+    });
+
+    counters[TOTAL] = totalCounter;
+};
+
+let getZombieType = (incomingHP) => {
+    let hp = parseInt(incomingHP);
+
+    let isNumeric = typeof hp === 'number';
```
Comment (ExSamaelSolbeg) [inline js/app.js]: please implement a solution without "typeof" keyword

File: js/app.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        counters[zombieType] = ++counter;
+
+        totalCounter++;
+    });
+
+    counters[TOTAL] = totalCounter;
+};
+
+let getZombieType = (incomingHP) => {
+    let hp = parseInt(incomingHP);
```
Comment (ExSamaelSolbeg) [inline js/app.js]: there are no requirements which are defines an integer value without decimal point
so, it's possible to get a zombie health value "10.5", "11.5" and so on

File: js/app.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    if (hp > 20) {
+        return STRONG;
+    }
+
+    return DEAD;
+};
+
+let outputData = () => {
```
Comment (ExSamaelSolbeg) [inline js/app.js]: same here
function is not required

File: js/app.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    for (let className in counters) {
+        let counter = counters[className];
```
Comment (ExSamaelSolbeg) [inline js/app.js]: please use "const" if a variable is not modifiable

File: js/app.js [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        let elementForCounter = document.getElementsByClassName(className);
+
+        if (elementForCounter.length) {
+            // Output to html
+            elementForCounter[0].innerHTML += '{' + counter + '}';
```
Comment (ExSamaelSolbeg) [inline js/app.js]: "+="
really?

Comment (ExSamaelSolbeg) [inline js/app.js]: please use a template string formatting instead of "+" concatenation

File: js/app.js [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+            // Output to console
+            console.log(elementForCounter[0].textContent);
```
Comment (ExSamaelSolbeg) [inline js/app.js]: it doesn't make sense to output html into console
the point of this task requirement is to compare console output and html output

**Review by ExSamaelSolbeg — CHANGES_REQUESTED** [review]

File: js/app.js (hunk 1 above)
Comment (Stereotip42) [reply js/app.js]: Fixed

File: js/app.js (hunk 2 above)
Comment (Stereotip42) [reply js/app.js]: fixed

File: js/app.js (hunk 3 above)
Comment (Stereotip42) [reply js/app.js]: fixed

File: js/app.js (hunk 4 above)
Comment (Stereotip42) [reply js/app.js]: fixed

File: js/app.js (hunk 5 above)
Comment (Stereotip42) [reply js/app.js]: fixed

File: js/app.js (hunk 6 above)
Comment (Stereotip42) [reply js/app.js]: fixed

File: js/app.js (hunk 7 above)
Comment (Stereotip42) [reply js/app.js]: fixed

File: js/app.js (hunk 8 above)
Comment (Stereotip42) [reply js/app.js]: fixed

File: js/app.js (hunk 9 above)
Comment (Stereotip42) [reply js/app.js]: fixed

Comment (Stereotip42) [reply js/app.js]: fixed, but template string formatting is missing in the PDF file of the lecture

File: js/app.js (hunk 10 above)
Comment (Stereotip42) [reply js/app.js]: fixed

**Review by ExSamaelSolbeg — APPROVED** [review]

**Pull request merged**